### DISCUSSION TITLE
user_id not registering when the foreign key is different (but still references the User model)

### DISCUSSION THREAD

**Discussion by simonelepore** [owen-it/laravel-auditing#960] [Q&A]
I have a posts table and an users table. Posts has a creator_id (which references the id on User). I tried to change the morph_prefix into "creator" but it still doesn't work. How do I handle this? Thank you.

**Comment by parallels999** [comment]
No idea how to reproduce this problem, little information, no examples

**Reply by simonelepore** [reply]
this is the relation in the Post Model

```
    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'creator_id');
    }
```

this is the column:

`            $table->foreignId('creator_id')->nullable()->constrained('users')->nullOnDelete();`

this is the config in audit.php


```
    'user'      => [
        'morph_prefix' => 'user',
        'guards'       => [
            'web',
            'api'
        ],
        'resolver'     => OwenIt\Auditing\Resolvers\UserResolver::class
    ],
```

the user_id in the audits table simply does not get registered when I submit a post. If you need further informations let me know and I'll provide them! thanks

**Reply by parallels999** [reply]
https://github.com/owen-it/laravel-auditing/blob/9357577514120015961e1ae90f0ee3b6d6193de6/src/Auditable.php#L347-L350

Morph prefix, and Post foreing key are two different things
I already tried it, the registry is audited correctly, look at `old_values`, `new_values`

**Reply by simonelepore** [reply]
yes, old_values and new_values are being stored, but the user_id isn't. what am I doing wrong?

**Reply by parallels999** [reply]
>what am I doing wrong?

Something, but no one can tell you without seeing the problem in a functional app

**Reply by simonelepore** [reply]
what's the best way for me to provide informations so I can receive help?

**Reply by parallels999** [reply]
>the problem in a functional app

Create a repo showing the problem, 2 commits, initial for fresh installation, and one for your customizations with the problem, or create a failing tests
